### DISCUSSION TITLE
UrlFormatter with two or more parameters

### DISCUSSION THREAD

**Discussion by teafmartins** [serenity-is/Serenity#7004] [Help]
Hello,

Is it possible to use urlformatter with 2 or more parameters? something like:

[UrlFormatter(UrlFormat = "~/Events/Event/list?customer={0}&product={1}")]

Thanks

**Comment by LSNicholls** [comment]
This sounds like an enhancement request, not a question about whether it is possible "now", right?  If it were going to work there would have to be a UrlProperty1, and I can see a bunch of problems in implementing it unless its default was specified to be the same as UrlProperty or an error was thrown if {1} was used and no UrlProperty1 was provided.

In the meantime I'm pretty sure you you can do it yourself... something like this:

`[UrlFormatter(UrlFormat = "~/Events/Event/list?",  UrlProperty="MyUrlArgument") ]`

... now you have to add a column to your row, MyUrlArgument, which fashions the single available argument the way you want.  For example, you could do   

`[DisplayName("MyUrlArgument")  , Expression("concat('customer=', CustomerId, '&product=', ProductId)")`
